### PR TITLE
[WFLY-11124] Reduce the dependencies used in the test suite. Instead …

### PR DESCRIPTION
**Pull request by jamezp**
…of relying on all server dependencies, be explicit with the dependencies required. Some dependencies are still a bit too tightly coupled to the server however there could be a reason for this beyond my knowledge of the test.

https://issues.jboss.org/browse/WFLY-11124 also relates to https://issues.jboss.org/browse/WFLY-10529.

This is a testsuite only change which attempts to reduce the server dependencies. Some tests still seem to require dependencies directly from subsystems which IMO seems wrong. I can file JIRA's for these if anyone agrees we should not be using these.

Tests should likely not be using any non-public API's/dependencies. That's out of the scope of this JIRA and PR, but we might want to open a further discussion on this.

### CODE REVIEW

File: testsuite/integration/smoke/pom.xml [hunk 1]
```diff
@@ -36,13 +36,43 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.ws.rs</groupId>
+            <artifactId>jboss-jaxrs-api_2.1_spec</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <!-- This is less than ideal, but required for some parsing tests.
+             See DataSourceOperationsUnitTestCase and AddMySqlDataSourceOperationsUnitTestCase
+        -->
```
Comment (bstansberry) [inline testsuite/integration/smoke/pom.xml]: This should be fixed for sure. (I don't mean in this PR though.)

Tests of model parsing should be done in the subsystem's own module.

IMO DsMgmtTestBase.marshalAndReparseDsResources, ResourceAdapterOperationsUnitTestCase.marshalAndReparseRaResources and AbstractMgmtTestBase.modelToXml should all be removed.

**Comment by bstansberry** [comment]
LGTM.

You've correctly made the point before that not reusing production code constants for test strings makes for better tests, since it means the test will catch incompatible changes in the constant. So just for that reason alone this is a good improvement.

File: testsuite/integration/smoke/pom.xml (hunk 1 above)
Comment (jamezp) [reply testsuite/integration/smoke/pom.xml]: Makes sense to me. I'll file an issue for this to get fixed later. I might file an issue for the other places I think should be fixed and the component lead can decide how feasible that is.

**Comment by jamezp** [comment]
I'm going to run a JDK 11 test run locally to see if I see the same issues on the JDK 10 run.

**Comment by jamezp** [comment]
Placing a hold on this because there does seem to be an issue with Java 11 and the WS tests that this PR introduced.

**Comment by jamezp** [comment]
Java 11 CI run https://ci.wildfly.org/viewLog.html?buildId=124591&buildTypeId=WF_MasterLinuxJdk11

**Review by bstansberry — APPROVED** [review]

**Pull request merged**